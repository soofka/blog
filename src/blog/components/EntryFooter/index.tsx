import * as React from 'react';

import { ContentDivider } from 'blog/components/ContentDivider';
import { ScrollTopButton } from 'blog/components/ScrollTopButton';
import { getEntryBaseUrl } from 'common/helpers';
import { EntryComments } from './components/EntryComments';


interface EntryFooterPropsInterface {
  title: string;
  niceUrl: string;
}

export const EntryFooter = (props: EntryFooterPropsInterface): JSX.Element => {
  const { title, niceUrl } = props;
  return (
    <div>
      <ScrollTopButton/>
      <ContentDivider/>
      <EntryComments
        title={title}
        url={`${getEntryBaseUrl()}${niceUrl}`}
        shortName={niceUrl}
        identifier={niceUrl}
      />
    </div>
  );
};
