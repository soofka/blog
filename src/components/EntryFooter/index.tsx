import * as React from 'react';

import EntryComments from './components/EntryComments';
import ContentDivider from 'components/ContentDivider';
import ScrollTopButton from 'components/ScrollTopButton';

interface EntryFooterPropsInterface {
  title: string;
  niceUrl: string;
  niceUrlWithBasePath: string;
}

export const EntryFooter = (props: EntryFooterPropsInterface): JSX.Element => {
  const { title, niceUrl, niceUrlWithBasePath } = props;
  return (
    <div>
      <ScrollTopButton/>
      <ContentDivider/>
      <EntryComments
        title={title}
        url={niceUrlWithBasePath}
        shortName={niceUrl}
        identifier={niceUrl}
      />
    </div>
  );
};

export default EntryFooter;
