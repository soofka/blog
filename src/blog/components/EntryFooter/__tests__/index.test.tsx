import { shallow } from 'enzyme';
import * as React from 'react';

import EntryFooter from 'components/EntryFooter';

describe('<EntryFooter/>', () => {

  it('renders correctly', () => {
    expect(shallow(<EntryFooter/>)).toMatchSnapshot();
  });

});
