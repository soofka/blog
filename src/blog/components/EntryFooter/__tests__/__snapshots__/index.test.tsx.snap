// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EntryFooter/> renders correctly 1`] = `
<div>
  <Component />
  <Component />
  <EntryComments />
</div>
`;
